### ISSUE TITLE
D401: Support GObject.Property from PyGObject

### ISSUE THREAD

**Issue by staticssleever668** [charliermarsh/ruff#3305]
* Ruff version: 3ed539d50ed6260358c97d2e00b49bad4abfa37e (0.0.253)

```py
from gi.repository import GObject


class Thingy(GObject.GObject):
    _beep = "boop"

    @GObject.Property
    def good_property(self):
        """This property method docstring does not need to be written in imperative mood."""
        return self._beep

my_thingy = Thingy()
print(my_thingy.good_property)
```

When run this as expected prints `boop`.
But `ruff check --isolated --select D401 test.py` gives a D401 warning: 

```
test.py:9:9: D401 First line of docstring should be in imperative mood: "This property method docstring does not need to be written in imperative mood."
```

I made a branch that removes this warning for GObject.Property (https://github.com/staticssleever668/ruff/tree/feat_support_gobject_property), but I'm not sure if support for third-party libraries in such cases makes sense.

As a note, pydocstyle produces a warning for this as well:
```
test.py:9 in public method `good_property`:
        D401: First line should be in imperative mood; try rephrasing (found 'This')
```

**Comment by staticssleever668** [comment]
Yes, it's good! :+1:

```toml
[tool.ruff.pydocstyle]
ignore-decorators=["gi.repository.GObject.Property"]
```

**Comment by charliermarsh** [comment]
I think the unimplemented `property_decorators` would be even better, since then we'd still lint the docstring, we just wouldn't enforce imperative style.
